src/main/shell-session/shell-env-modifier/shell-env-modifier-registration.ts

```diff
@@ -0,0 +1,14 @@
+/**
+ * Copyright (c) OpenLens Authors. All rights reserved.
+ * Licensed under MIT License. See LICENSE in root directory for more information.
+ */
+
+import type { noop } from "../../../common/utils";
+
+export interface ShellEnvClusterData {
+  context: string;
+  kubeconfigPath: string;
+  id: string;
+}
+
+export type ShellEnvModifier = ((cluster: ShellEnvClusterData, env: Record<string, string | undefined>) => Record<string, string | undefined>) | typeof noop;
```

Comment:
I think that having `typeof noop` is wrong here, I would just make the field in `LensMainExtension` be optional.